File: 240202/challenge111.py
import RPi.GPIO as GPIO

# GPIO 핀 번호 설정
led_pin = 4  # 라즈베리 파이의 0번 핀을 사용

# GPIO 설정
GPIO.setmode(GPIO.BCM)
GPIO.setup(led_pin, GPIO.OUT)

try:
    while True:
        # 사용자로부터 입력 받기
        user_input = input("키를 입력하세요 (N: LED 켜기, F: LED 끄기, Q: 종료): ")

        # 입력에 따라 LED 켜고 끄기
        if user_input.upper() == 'N':
            GPIO.output(led_pin, GPIO.HIGH)  # LED 켜기
        elif user_input.upper() == 'F':
            GPIO.output(led_pin, GPIO.LOW)   # LED 끄기
        elif user_input.upper() == 'Q':
            break  # 프로그램 종료
        else:
            print("잘못된 입력입니다. 다시 시도하세요.")

except KeyboardInterrupt:
    pass
finally:
    # GPIO 리소스 정리
    GPIO.cleanup()

File: 240202/challenge111.c
#include <wiringPi.h>
#include <stdio.h>

#define LED_PIN 4

int main(void) {
    wiringPiSetup();
    pinMode(LED_PIN, OUTPUT);

    char userInput;

    while (1) {
        printf("키를 입력하세요 (N: LED 켜기, F: LED 끄기, Q: 종료): ");
        scanf(" %c", &userInput);

        if (userInput == 'N' || userInput == 'n') {
            digitalWrite(LED_PIN, HIGH);  // LED 켜기
        } else if (userInput == 'F' || userInput == 'f') {
            digitalWrite(LED_PIN, LOW);   // LED 끄기
        } else if (userInput == 'Q' || userInput == 'q') {
            break;  // 프로그램 종료
        } else {
            printf("잘못된 입력입니다. 다시 시도하세요.\n");
        }
    }

    return 0;
}
